### PR TITLE
Fix return value of "changed" in check_mode for all modules

### PR DESCRIPTION
**Pull request by jake2184**
##### SUMMARY
The plugins within this module do not currently support 'changed' in check mode correctly. This is not suitable for an enterprise storage solution such as FlashBlade. I have a basic initial setup, yet when I run Ansible in check mode I am met by a whole host of yellow changes. This PR fixes that. 

The logic already exists in pretty much every case, this PR is a case of shuffling where the `if not module.check_mode` is positioned, and then altering `changed` value.

This was picked up and highlighted in #110, where I was told support for this is some time away - as this PR shows, it is not a great deal of work to fix this, and is a very important aspect of Ansible modules, _especially_ for something as critical as a network storage system.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
All plugins

##### ADDITIONAL INFORMATION
I have tested this against my Pure with some, but not all of the modules. Actively tested:
- purefb_alert
- purefb_dns
- purefb_ds
- purefb_dsrole
- purefb_fs
- purefb_policy
- purefb_smtp

### CODE REVIEW

**Review by sdodsley — COMMENTED** [review]
This looks good - thanks for addressing this.
Will check the modules out

**Comment by jake2184** [comment]
I'm make changes in a local copy - if you let me know when you've finished reviewing/adding comments, I'll batch up all changes into a single commit to keep things cleaner.

**Comment by jake2184** [comment]
Bump on this @sdodsley

**Comment by sdodsley** [comment]
@jake2184 I think we are good. If you push your local changes I will mark this for review by the wider community.

**Comment by jake2184** [comment]
Pushed

**Review by sdodsley — APPROVED** [review]
Let's get at least one more Approval before we merge this

**Review by dnix101 — DISMISSED** [review]
lgtm

**Review by genegr — APPROVED** [review]
shipit

File: collections/ansible_collections/purestorage/flashblade/plugins/modules/purefb_bucket_replica.py [hunk 1]
```diff
@@ -162,11 +162,11 @@ def create_rl(module, blade, remote_cred):
...
     """Update Bucket Replica Link"""
-    changed = True
-    if not module.check_mode:
-        changed = False
-        new_cred = local_replica_link.remote.name + '/' + module.params['credential']
-        if local_replica_link.paused != module.params['paused'] or local_replica_link.remote_credentials.name != new_cred:
+    changed = False
+    new_cred = local_replica_link.remote.name + '/' + module.params['credential']
+    if local_replica_link.paused != module.params['paused'] or local_replica_link.remote_credentials.name != new_cred:
+        changed = True
```
[the remark this answers is not in the document]
Comment (sdodsley) [reply collections/ansible_collections/purestorage/flashblade/plugins/modules/purefb_bucket_replica.py:168]: Dealt with

File: collections/ansible_collections/purestorage/flashblade/plugins/modules/purefb_certgrp.py [hunk 2]
```diff
@@ -112,38 +112,42 @@ def create_certgrp(module, blade):
...
+        for new_cert in range(0, len(module.params['certificates'])):
+            certificate = module.params['certificates'][new_cert]
+            if certificate in current:
+                if module.params['state'] == 'absent':
+                    changed = True
+                    if not module.check_mode:
                         try:
                             blade.certificate_groups.remove_certificate_group_certificates(certificate_names=[certificate],
                                                                                            certificate_group_names=[module.params['name']])
                             changed = True
```
[the remark this answers is not in the document]
Comment (sdodsley) [reply collections/ansible_collections/purestorage/flashblade/plugins/modules/purefb_certgrp.py]: Dealth with

**Comment by jake2184** [comment]
Thanks for reviewing and merging.

Do you have an eta/schedule for release of the next version (v1.5.0?) that will include this?

**Comment by sdodsley** [comment]
Not yet - shouldn't be long though - definitely before Xmas

**Pull request merged**